packages/rum-core/src/performance-monitoring/capture-hard-navigation.js

```diff
@@ -182,17 +179,21 @@ function captureHardNavigation(transaction) {
     // must be zero otherwise the calculated relative _start time would be wrong
     transaction._start = 0
 
-    createNavigationTimingSpans(timings, timings.fetchStart).forEach(function(
-      span
-    ) {
-      if (isValidSpan(transaction, span)) {
-        span.traceId = transaction.traceId
```

Comment:
@jahtalab Curious Why do we do this only for navigation timing span, why not for resource timing?

Reply:
I think it's a bug if we don't, would you please create an issue for it?

Reply:
Yeah good catch, Lets remove both trace_id and sampled for nav timing then? Because sampled is not even sent as part of the payload.

Reply:
One of the test was utlizing it, we can remove it in another PR.

Reply:
I think we should make sure that spans have complete trace context (`traceId`,`parentId` and `sampled`) the reason for this is that users might decide to use this information in their logs or otherwise transfer the trace context to another application.

Reply:
Yeah makes sense, I will add the `sampled` flag to the createPayload method and remove it from here once the PR is merged. It is the only field that is missing from all the spans.

Reply:
I create https://github.com/elastic/apm-agent-rum-js/issues/307 to keep track of this.